libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1249,14 +1253,24 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		templ("returnSize", "returndatasize()");
 	else
 		templ("returnSize", to_string(retSize));
+
+	templ("reservedReturnSize", dynamicReturnSize ? "0" : to_string(retSize));
```

Comment:
Maybe it would make sense to rename `retSize` to `estimatedReturnSize` or `expectedReturnSize`.